src/rdiffbackup/locations/repository.py

```diff
@@ -35,10 +35,69 @@
 )
 
 
-class Repo():
+class Repo(locations.Location):
     """
     Represent a Backup Repository as created by rdiff-backup
     """
+    def __init__(self, base_dir, log, force, must_be_writable, must_exist,
+                 create_full_path=False, can_be_sub_path=False):
+        """
+        Initialize the repository class
+
+        can_be_sub_path is True if the base_dir can actually be a repository,
+        but also a sub-directory or even an increment file, mostly used for
+        restore actions.
+        """
+        # if the base_dir can be a sub-file, we need to identify the actual
+        # base directory of the repository
+        if can_be_sub_path:
+            self.orig_path = base_dir
+            (base_dir, restore_index, restore_type) = \
+                base_dir.get_repository_dirs()
+            self.restore_index = restore_index
+            self.restore_type = restore_type
+            # we delay the error handling until the check step
+        else:
+            self.restore_index = None
+            self.restore_type = None
+
+        # Finish the initialization with the identified base_dir
+        super().__init__(base_dir, log, force)
+        self.create_full_path = create_full_path
+        self.must_be_writable = must_be_writable
+        self.must_exist = must_exist
+        self.can_be_sub_path = can_be_sub_path
+        self.data_dir = self.base_dir.append_path(b"rdiff-backup-data")
+        self.incs_dir = self.data_dir.append_path(b"increments")
+
+    def check(self):
+        if self.can_be_sub_path and self.restore_type is None:
+            # there is nothing to save, the user must first give a correct path
+            return 1
```

Comment:
Because at the end I use the result to return it as the program's return code (see the action's check/setup method, followed by the Main._main_run function). I currently don't feel the need to have more than 0 or 1, but I could introduce other error codes (and I might).
From what I know, Exceptions are a double-edged sword: it's simple and "definitive", but it also has a performance impact, it makes it more difficult to debug, and gives you not so nicely formatted messages. Personally, I prefer to reserve exceptions for when I really need. As I paid attention to have the check/setup functions properly handled, I don't need it.

Reply:
To clarify this point, maybe we should consider declaring constant "SUCCESS = 0" and "ERROR = 1" ?